natlas-server/app/__init__.py

```diff
@@ -130,8 +130,12 @@ def create_app(config_class=Config, load_config=False):
 				print("AgentScript table doesn't exist yet. Ignore if flask db upgrade.")
 
 		# Grungy thing so we can use flask db and flask shell before the config items are initially populated
-		if "ELASTICSEARCH_URL" in app.config:
-			app.elastic = Elastic(app.config['ELASTICSEARCH_URL'])
+		if "ELASTICSEARCH_URL" in os.environ:
```

Comment:
Please do not search os.environ directly outside of the config file. If you want to be able to set the ELASTICSEARCH_URL in the environment, it needs to be done through the application config. All environment variables, with very few exceptions, should be referenced from the Config, not from anywhere else in the code.

Reply:
I tryed various ways to get this to load in via the config.py, but it didnt seem to work. I might have to get some help with understanding the prepare_defaults stuff.

Reply:
prepare_defaults just inserts into the database for whatever the default database-stored values are. I think in most deployment situations it would be easier to pre-populate this with a better config option than using the web interface to change the url it's pointing to. Once there is an api for modifying these config options then it'll be easier, but in the meantime, loading via environment variable is probably very handy. You should basically just copy another config option in config.py and then change it to ELASTICSEARCH_URL instead of whatever it is. Since it's going to be a string you don't need to do any special parsing. Then it's available in app.config or current_app.config, depending on the context you're in. (In the __init__ script, it's app.config)

But I'm not sure if it's as easy as just adding the environment variable or if it will be loaded from the database, overriding the env variable. We might have to add a check when populating the config that if it's in the environment variables then to ignore it from the database. I don't really like that idea in the long term because it introduces confusion, but until we have a proper api that can be automated against, it might be necessary.